Daily video calls

lunchpaillola opened the pull request:
Adding Daily Video Calls to Calendso. Here's a demo of what creating and joining a video room looks like 

`https://www.loom.com/share/f1fe710a081a46078621749d532a6e1c```

lunchpaillola committed:
modifying how daily emails send

amirhmoradi commented:
Hi @lunchpaillola , thanks for the great work. I have some feedback if you don't mind :)

I believe we could use already available "isdedicated" concept and avoid hard coding an external provider (ie Daily here) in the event manager (and other places).

For the  db side, I would also suggest having a separate table to hold Daily's info and reference the booking id. Core tables shall theoretically be clean of external providers, to allow for better future evolutions.

About migrations, it would be best to have just one migration per PR. You could prepare it by going from a base DB (from latest main branch) and the applying your schema and then generating the migration)

For the emails templates, we could condition the meeting username and password existence and so avoid the hard coded provider condition.

Also, I would suggest having another PR for code indentation fixes and lib/deps updates.

I would love to see your PR merged soon :)

lunchpaillola commented:
> Hi @lunchpaillola , thanks for the great work. I have some feedback if you don't mind :)
> 
> I believe we could use already available "isdedicated" concept and avoid hard coding an external provider (ie Daily here) in the event manager (and other places).
> 
> For the db side, I would also suggest having a separate table to hold Daily's info and reference the booking id. Core tables shall theoretically be clean of external providers, to allow for better future evolutions.
> 
> About migrations, it would be best to have just one migration per PR. You could prepare it by going from a base DB (from latest main branch) and the applying your schema and then generating the migration)
> 
> For the emails templates, we could condition the meeting username and password existence and so avoid the hard coded provider condition.
> 
> Also, I would suggest having another PR for code indentation fixes and lib/deps updates.
> 
> I would love to see your PR merged soon :)

Thanks for your review and the comments!

lunchpaillola committed:
migration updates

lunchpaillola committed:
updating cal will provide a zoom meeting url

lunchpaillola committed:
modifying how daily emails send

lunchpaillola committed:
daily table

lunchpaillola committed:
updating cal will provide a zoom meeting url

lunchpaillola committed:
daily table

lunchpaillola committed:
rebasing updates

lunchpaillola committed:
updating Daily references to a new table

lunchpaillola committed:
updating internal notes

lunchpaillola committed:
merge updates, adding Daily references to book/events.ts

lunchpaillola committed:
updated video email templates to remove Daily specific references

dependabot[bot] committed:
:arrow_up: Bump tailwindcss from 2.2.14 to 2.2.15

Bumps [tailwindcss](https://github.com/tailwindlabs/tailwindcss) from 2.2.14 to 2.2.15.
- [Release notes](https://github.com/tailwindlabs/tailwindcss/releases)
- [Changelog](https://github.com/tailwindlabs/tailwindcss/blob/master/CHANGELOG.md)
- [Commits](https://github.com/tailwindlabs/tailwindcss/compare/v2.2.14...v2.2.15)

---
updated-dependencies:
- dependency-name: tailwindcss
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

lunchpaillola committed:
updating the events.ts and refactoring in the event manager

lunchpaillola committed:
removing the package-lock

lunchpaillola committed:
changing calendso video powered by Daily.co to cal video powered by Daily.co

lunchpaillola committed:
updating some of the internal Daily notes

lunchpaillola committed:
added a modal for when the call/ link is invalid

lunchpaillola committed:
removing handle errors raw from the Daily video client

lunchpaillola committed:
prettier formatting fixes

lunchpaillola committed:
Merge pull request #11 from Lunch-Pail-Labs/dependabot/npm_and_yarn/tailwindcss-2.2.15

⬆️ Bump tailwindcss from 2.2.14 to 2.2.15

lunchpaillola committed:
Merge branch 'calendso:main' into main

baileypumfleet commented on .husky/pre-commit:
```diff
@@ -2,3 +2,6 @@
 yarn lint-staged
+npm test
+npm test
+initial test
```
Can we remove these tests from the pre-commit hook? It slows commits down dramatically and throws an error on my machine due to initial not being found? Testing is run once a PR is opened anyway

baileypumfleet commented on lib/videoClient.ts:
```diff
@@ -70,7 +70,7 @@ const zoomAuth = (credential) => {
         });
         credential.key.access_token = responseBody.access_token;
-        credential.key.expires_in = Math.round(+new Date() / 1000 + responseBody.expires_in);
+        credential.key.expires_in = Math.round(+new Date() / 1000 + responseBody.expires_in) * 1000;
```
What does this change do, as I'm not familiar with how the video integrations work?

lunchpaillola replied:
It changes the credential.key.expires to milliseconds instead of seconds. It could be just me, but the token kept reading as expired for me when credential.key.expires_in was in seconds, and I couldn't use Zoom.

lunchpaillola committed:
Merge branch 'main' of https://github.com/calendso/calendso

lunchpaillola committed:
Added the Daily location to calendar events and updated Cal video references to Daily.co video

lunchpaillola committed:
updating references to create in event manager to check for Daily video

lunchpaillola committed:
fixing spacing on the cancel booking modal and adding Daily references in the event manager

lunchpaillola committed:
Merge branch 'calendso:main' into main

lunchpaillola committed:
formatting fixes

pumfleet reviewed (approved):
Code looks good to me, will do some local testing now.

zomars commented on .env.example:
```diff
@@ -22,6 +22,9 @@ MS_GRAPH_CLIENT_SECRET=
 ZOOM_CLIENT_SECRET=
 
+#Used for the Daily integration
+DAILY_API_KEY=
```
This is awesome! @lunchpaillola 

Would you kindly add instructions on how to setup the integration like the other ones in the `README.md`?

lunchpaillola replied:
Definitely, looks like I accidentally removed that from some previous commits but it's up now!

zomars commented on .eslintrc.json:
```diff
@@ -16,7 +16,12 @@
+    "prettier/prettier": [
+      "error",
+      {
+        "endOfLine": "auto"
```
Is this fixing something specific?

lunchpaillola replied:
thanks for the comment,I think I was getting some prettier errors when committing a while ago without the end of line auto but it can definitely be removed.

zomars replied:
Let's remove this for now, we don't know how is gonna affect the rest of the files.

lunchpaillola replied:
got it! should be removed now

zomars reviewed (changes requested):
This is awesome! @lunchpaillola

Would you kindly add instructions on how to setup the integration like the other ones in the README.md?

EDIT: I've stumbled upon this error when trying to test it locally:

![image](https://user-images.githubusercontent.com/3504472/135672790-d4e85868-00d7-4ecb-afed-2b74d83e6e50.png)

EDIT 2: Also, can we type cast this so we know what properties to expect from the response?  

![image](https://user-images.githubusercontent.com/3504472/135673048-d3dcdec4-e4c8-4a8b-834e-54bd822403f1.png)

lunchpaillola committed:
Merge branch 'calendso:main' into main

lunchpaillola committed:
updating the readme file

zomars commented on pages/api/book/event.ts:
```diff
@@ -439,6 +440,44 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
+  if (isDaily) {
+    const response = await fetch("https://api.daily.co/v1/meeting-tokens", {
+      method: "POST",
+      body: JSON.stringify({ properties: { room_name: dailyEvent.name, is_owner: true } }),
```
This line is throwing an error when testing locally:

![](https://user-images.githubusercontent.com/3504472/135672790-d4e85868-00d7-4ecb-afed-2b74d83e6e50.png)

lunchpaillola replied:
Hi @zomars 

Thanks for sharing! I'm trying to recreate this error locally. Are you still able to book or does it completely not allow you to create a booking?

zomars replied:
Let me try to re-test

zomars replied:
I couldn't replicate the error anymore

PeerRich replied:
was this resolved?

lunchpaillola replied:
Thanks for the comments everyone! In the latest commit, I added an interface for the daily result and updated the daily event to resolve the type error. I think it runs regardless but this should be a little cleaner.

lunchpaillola committed:
adding a daily interface in the event manager

lunchpaillola committed:
Merge branch 'calendso:main' into main

lunchpaillola committed:
adding daily to the location labels

zomars commented:
All seems ok so far. Would you please check out merge conflict and run `yarn` to update the lockfile? It seems that is not up to date.

lunchpaillola committed:
Merge branch 'calendso:main' into main

lunchpaillola committed:
added a note to cal event parser

lunchpaillola committed:
resolving yarn merge conflicts

zomars commented on lib/events/EventManager.ts:
```diff
@@ -91,13 +105,25 @@ export default class EventManager {
+          meetingPassword: result.videoCallData?.password,
+          meetingUrl: result.videoCallData?.url,
+        };
+      }
+      if (!isDailyResult) {
```
Why not just an `else` ?

```suggestion
      } else {
```

zomars replied:
Or skip the `else` altogether since we're using `return`.

zomars commented on lib/events/EventManager.ts:
```diff
@@ -232,8 +262,12 @@ export default class EventManager {
     const credential = this.getVideoCredential(event);
 
-    if (credential) {
+    const isDaily = event.location === "integrations:daily";
```
I see we're using this string everywhere. Maybe we can extract this into a `CONSTANT` ?

zomars commented on pages/api/book/event.ts:
```diff
@@ -20,6 +20,13 @@ import prisma from "@lib/prisma";
 import { BookingCreateBody } from "@lib/types/booking";
 import { getBusyVideoTimes } from "@lib/videoClient";
 
+export interface dailyReturn {
```
```suggestion
export interface DailyReturnType {
```

lunchpaillola replied:
sure! I can make that update

zomars reviewed (commented):
Looking good, will re-test later today.

lunchpaillola committed:
updating dailyReturn to DailyReturnType

zomars commented:
Testing:
With integration enabled
- [ ] Creating Bookings
- [ ] Creating Event Types
- [ ] Emails are being sent
- [ ] Can cancel
- [ ] Can reschedule
- [ ] Can join call

With integration disabled
- [ ] Creating Bookings
- [ ] Creating Event Types
- [ ] Emails are being sent
- [ ] Can cancel
- [ ] Can reschedule

zomars commented on pages/api/book/event.ts:
```diff
@@ -350,8 +357,9 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
-      const videoBusyTimes = await getBusyVideoTimes(credentials);
+      const videoBusyTimes = (await getBusyVideoTimes(credentials)).filter((time) => time);
       calendarBusyTimes.push(...videoBusyTimes);
+      console.log("calendarBusyTimes==>>>", calendarBusyTimes);
```
This is throwing an error when trying to book a regular Zoom booking with Daily disabled (DAILY_API_KEY is undefined)

![image](https://user-images.githubusercontent.com/3504472/136072549-1c3886fd-23b1-47a9-9ce3-c5de53966444.png)

zomars replied:
Actually never mind. It seems that's a separate issue with Zoom. You can disregard this. All good now.

lunchpaillola committed:
removing prettier auto and refactoring integrations: daily in the event manager

zomars commented on .eslintrc.json:
```diff
@@ -16,7 +16,6 @@
   ],
   "plugins": ["@typescript-eslint", "prettier", "react", "react-hooks"],
   "rules": {
-    "prettier/prettier": ["error"],
```
This should not be removed thought 😅

lunchpaillola replied:
oops, I undid that change now so it should match the eslintrc.json that's currently in the main branch! Thanks for flagging.

zomars commented on .eslintrc.json:
```diff
@@ -16,7 +16,6 @@
   ],
   "plugins": ["@typescript-eslint", "prettier", "react", "react-hooks"],
   "rules": {
```
```suggestion
  "rules": {
      "prettier/prettier": ["error"],
```

lunchpaillola committed:
removing changes to estlintrc.json

zomars reviewed (approved):
Thank you for your patience @lunchpaillola 🙏🏽 

I think we're good to go!

zomars commented on README.md:
```diff
@@ -273,6 +273,12 @@ Contributions are what make the open source community such an amazing place to b
 12. Click "Done".
 13. You're good to go. Now you can easily add your Zoom integration in the Cal.com settings.
 
+## Obtaining Daily API Credentials
+1. Open [Daily](https://www.daily.co/) and sign into your account. 
+2. From within your dashboard, go to the [developers](https://dashboard.daily.co/developers) tab.
+3. Copy your API key.
+4. Now paste the API key to your .env file into the `DAILY_API_KEY` field in your .env file.
+
```
```suggestion
## Obtaining Daily API Credentials

1. Open [Daily](https://www.daily.co/) and sign into your account.
2. From within your dashboard, go to the [developers](https://dashboard.daily.co/developers) tab.
3. Copy your API key.
4. Now paste the API key to your .env file into the `DAILY_API_KEY` field in your .env file.

```

zomars replied:
One last change to please the linter @lunchpaillola

lunchpaillola replied:
That should be updated now! Let me know if you need anything else @zomars

zomars replied:
Thank you. Let's just wait for the linter

lunchpaillola committed:
updating read me formatting

lunchpaillola committed:
indent space for Daily ReadMe section

lunchpaillola committed:
Merge branch 'calendso:main' into main

lunchpaillola committed:
resolving the merge conflicts in the yarn file
